sdks/python/apache_beam/runners/portability/spark_runner.py

```diff
@@ -41,27 +43,23 @@ def run_pipeline(self, pipeline, options):
     return super(SparkRunner, self).run_pipeline(pipeline, options)
 
   def default_job_server(self, options):
-    # TODO(BEAM-8139) submit a Spark jar to a cluster
+    spark_options = options.view_as(pipeline_options.SparkRunnerOptions)
+    if (spark_options.spark_submit_uber_jar
+        and not re.match(LOCAL_MASTER_PATTERN, spark_options.spark_master_url)):
+      if sys.version_info < (3, 6):
+        raise ValueError(
+            'spark_submit_uber_jar requires Python 3.6+, current version %s'
+            % sys.version)
+      return spark_uber_jar_job_server.SparkUberJarJobServer(
+          re.sub('^spark://', 'http://', spark_options.spark_master_url),
```

Comment:
The master and rest endpoints are distinct, so I should make the rest url an independent pipeline option before merging this PR.